tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/GenerateTimeline.scala

```diff
@@ -27,7 +27,9 @@ abstract class TimelineTiming(
     val endTime: Long)
 
 class TimelineTaskInfo(val stageId: Int, val taskId: Long,
-    startTime: Long, endTime: Long, val duration: Long)
+    startTime: Long, endTime: Long, val duration: Long,
```

Comment:
With all the `val` parameters, should this just be a `case class`?

Reply:
`TimelineTiming` would then have to change from an abstract class to a trait. I can do it, but it does not feel directly related to the issue being dealt with.